airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -325,10 +325,20 @@ def create_legacy_to_per_partition_state_migration(
             raise ValueError(
                 f"LegacyToPerPartitionStateMigrations can only be applied on a DeclarativeStream with a SimpleRetriever. Got {type(declarative_stream.retriever)}"
             )
-        if not isinstance(declarative_stream.retriever.partition_router, SubstreamPartitionRouterModel):
+
+        partition_router = declarative_stream.retriever.partition_router
+        router_class = partition_router[0].get("class_name") if partition_router else None
+
+        is_default_substream_partition_router = isinstance(partition_router, SubstreamPartitionRouterModel)
+        is_custom_substream_partition_router = router_class == SubstreamPartitionRouterModel and isinstance(
```

Comment:
I'm not sure I get this check. would the router_class also be of type `SubstreamPartitionRouterModel` when it's custom?

I think we just need to check that the custom class has an attribute `parent_stream_config` as a proxy for verifying it'll work with the migration

Reply:
But `router_class` can be `ListPartitionRouter` as well, so I believe we need check `router_class` type

Added `is_parent_stream_config` validation